core/server/common/src/main/java/alluxio/RpcUtils.java

```diff
@@ -28,60 +28,152 @@
+   * @param description the format string of the description, used for logging
+   * @param args the arguments for the description
    * @param <T> the return type of the callable
    * @return the return value from calling the callable
    * @throws AlluxioTException if the callable throws an exception
    */
-  public static <T> T call(Logger logger, RpcCallable<T> callable)
-      throws AlluxioTException {
-    return call(logger, callable, true);
+  public static <T> T call(Logger logger, RpcCallable<T> callable, String methodName,
+      String description, Object... args) throws AlluxioTException {
+    return call(logger, callable, methodName, false, description, args);
   }
 
   /**
    * Calls the given {@link RpcCallable} and handles any exceptions thrown.
    *
-   * The logAnyFailure parameter indicates whether or not AlluxioExceptions are expected results
-   * (for example it would be false for the exists() call). In this case, we do not log the failure
-   * or increment failure metrics. When a RuntimeException is thrown, we always treat it as a
-   * failure and log an error and increment metrics.
+   * The failureOk parameter indicates whether or not AlluxioExceptions are expected results (for
+   * example it would be false for the exists() call). In this case, we do not log the failure or
+   * increment failure metrics. When a RuntimeException is thrown, we always treat it as a failure
+   * and log an error and increment metrics.
    *
    * @param logger the logger to use for this call
    * @param callable the callable to call
-   * @param logAnyFailure whether to log or increment metrics whenever the RPC fails
+   * @param methodName the name of the method, used for metrics
+   * @param failureOk whether failures are expected (affects logging and metrics)
+   * @param description the format string of the description, used for logging
+   * @param args the arguments for the description
    * @param <T> the return type of the callable
    * @return the return value from calling the callable
    * @throws AlluxioTException if the callable throws an exception
    */
-  public static <T> T call(Logger logger, RpcCallable<T> callable, boolean logAnyFailure)
-      throws AlluxioTException {
-    try (Timer.Context ctx = MetricsSystem.timer(getQualifiedMetricName(callable)).time()) {
-      logger.debug("Enter: {}", callable);
-      T ret = callable.call();
-      logger.debug("Exit (OK): {}", callable);
-      return ret;
+  public static <T> T call(Logger logger, RpcCallable<T> callable, String methodName,
+      boolean failureOk, String description, Object... args) throws AlluxioTException {
+    try (Timer.Context ctx = MetricsSystem.timer(getQualifiedMetricName(methodName)).time()) {
+      if (logger.isDebugEnabled()) { // avoid string format for better performance if debug is off
+        logger.debug("Enter: {}", String.format(description, (Object[]) args));
```

Comment:
I tested and this will simply work because `public static String format(String format, Object... args)`
```java
  public static void call(Logger logger, String methodName,
     boolean failureOk, String description, Object... args) throws AlluxioTException {
    logger.debug("Enter: {}", String.format(description, args)); 
  }
```

Reply:
to make the code clean, how about:
```
String des = (logger.isDebugEnabled()?) {String.format(description, (Object[]) args) : null;
```
and then simply have the following code?
```java
logger.debug("Enter: {}", des);
logger.debug("Exit (OK): {}", des);
logger.debug("Exit (Error): {}", des);
```

Reply:
For the first comment, I think it will be confused since it could be a single array object or intended to be one argument for each element in the array?
Sure, I change the format for the second comment.